### PR TITLE
clang-format: prefer clang-format-14

### PR DESCRIPTION
**Pull request by jasonish**
If reformatting all the code, move to a newer version of clang-format.

### CODE REVIEW

**Comment by jasonish** [comment]
Sorry for the edits, `gh pr create --fill` automatically submits as well!

File: scripts/clang-format.sh [hunk 1]
```diff
@@ -560,9 +560,13 @@ SetTopLevelDir
 
 RequireProgram GIT git
 # ubuntu uses clang-format-{version} name for newer versions. fedora not.
-RequireProgram GIT_CLANG_FORMAT git-clang-format-11 git-clang-format-10 git-clang-format-9 git-clang-format
+RequireProgram GIT_CLANG_FORMAT git-clang-format-14 git-clang-format-11 git-clang-format-10 git-clang-format-9 git-clang-format
 GIT_CLANG_FORMAT_BINARY=clang-format
-if [[ $GIT_CLANG_FORMAT =~ .*git-clang-format-11$ ]]; then
+if [[ $GIT_CLANG_FORMAT =~ .*git-clang-format-14$ ]]; then
```
[the remark this answers is not in the document]
Comment (jasonish) [reply scripts/clang-format.sh:565]: I don't think we should go greater than what we consider our "standard". For example, clang-format-17 and clang-format-14 don't seem to agree on something things.  But this type of versioning seems to be an ubuntu thing here? So maybe we should add 17 and down, and just ensure that our CI job sticks to a specific version.  14 was chosen as that is what you get with `apt install clang-format` on 22.04.

**Comment by catenacyber** [comment]
I love this :-)
As I had difficulties making clang-format-9 happy with my clang-14 or 15

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #10043, thanks!

**Pull request closed without merging**